File: packages/openland-api/fragments/MatchmakingRoom.ts
import gql from 'graphql-tag';

export const MatchmakingRoomFragment = gql`
    fragment MatchmakingRoomFragment on MatchmakingRoom {
        enabled
        questions {
            ... on TextMatchmakingQuestion {
                id
                title
                subtitle
            }
            ... on MultiselectMatchmakingQuestion {
                id
                title
                subtitle
                tags
            }
        }
        profiles {
            chatCreated
            user {
                id
                name
                photo
                primaryOrganization {
                    id
                    name
                    photo
                }
            }
            answers {
                ... on TextMatchmakingAnswer {
                    question {
                        id
                        title
                        subtitle
                    }
                    answer
                }
                ... on MultiselectMatchmakingAnswer {
                    question {
                        id
                        title
                        subtitle
                    }
                    tags
                }
            }
        }
    }
`;

File: android/app/src/main/java/com/openland/react/graphql/operations/queries/MatchmakingRoom.kt
package com.openland.react.graphql

import com.openland.spacex.*
import com.openland.spacex.gen.*
import org.json.*

internal val MatchmakingRoomSelector = obj(
            field("matchmakingRoom", "matchmakingRoom", arguments(fieldValue("peerId", refValue("peerId"))), obj(
                    field("__typename", "__typename", notNull(scalar("String"))),
                    fragment("MatchmakingRoom", MatchmakingRoomFragmentSelector)
                ))
        )
val MatchmakingRoom = object: OperationDefinition {
    override val name = "MatchmakingRoom"
    override val kind = OperationKind.QUERY
    override val body = "query MatchmakingRoom(\$peerId:ID!){matchmakingRoom(peerId:\$peerId){__typename ...MatchmakingRoomFragment}}fragment MatchmakingRoomFragment on MatchmakingRoom{__typename enabled questions{__typename ... on TextMatchmakingQuestion{__typename id title subtitle}... on MultiselectMatchmakingQuestion{__typename id title subtitle tags}}myProfile{__typename ...MatchmakingProfileFragment}profiles{__typename ...MatchmakingProfileFragment}}fragment MatchmakingProfileFragment on MatchmakingProfile{__typename chatCreated user{__typename id isYou name photo isBot primaryOrganization{__typename id name}}answers{__typename ... on TextMatchmakingAnswer{__typename question{__typename id title subtitle}answer}... on MultiselectMatchmakingAnswer{__typename question{__typename id title subtitle}tags}}}"
    override val selector = MatchmakingRoomSelector
}